src/daemon/rpmostree-sysroot-upgrader.c

```diff
@@ -962,6 +962,12 @@ perform_local_assembly (RpmOstreeSysrootUpgrader *self,
                                            cancellable, error))
     return FALSE;
 
+  /* Ensure we aren't holding any references to the tmpdir now that we're
+   * done. */
```

Comment:
Hmm, can you expand on this? We're already clearing the context in the destructor; do we need it earlier than that?

Reply:
Yes, but `rpmostree_sysroot_upgrader_deploy()` calls `rpmostree_syscore_write_deployment()` which calls `rpmostree_syscore_cleanup()` which nukes the tmpdir while the upgrader is still alive.

Reply:
Ahhh, gotcha. Can you mention that in the comment for posterity?

Reply:
Done :arrow_dow